### PR TITLE
Fix rename_all() on grouped data

### LINKED ISSUE #2947: Scoped rename functions don't work on grouped_df
In dplyr 0.7.1 these work:
```
mtcars %>% rename_all(toupper)
mtcars %>% group_by(vs) %>% rename(VS = vs)
```
But these don't (throw an error `All arguments must be named`):
```
mtcars %>% group_by(vs) %>% rename_all(toupper)
mtcars %>% group_by(vs) %>% rename_if(rlang::is_integerish, toupper)
mtcars %>% group_by(vs) %>% rename_at("mpg", toupper)
```

### PR DESCRIPTION
**Pull request by lionel-**
Closes #2947.

@hadley currently `rename_all()` does not rename the grouping variables. This is consistent with `mutate_all()` and `summarise_all()` but it seems that in the case of names it'd make sense to operate on the grouping vars as well. What do you think?

### CODE REVIEW

**Comment by hadley** [comment]
Can you figure out what's failing?

**Comment by lionel-** [comment]
The failure is unrelated to this PR as far as I can tell.

File: tests/testthat/test-colwise-select.R [hunk 1]
```diff
@@ -71,3 +71,9 @@ test_that("select_if() handles quoted predicates", {
   expect_identical(select_if(mtcars, "is_integerish"), expected)
   expect_identical(select_if(mtcars, ~is_integerish(.x)), expected)
 })
+
+test_that("rename_all() works with grouped data", {
```
Comment (krlmlr) [inline tests/testthat/test-colwise-select.R]: Adding a reference to the GH issue or PR in the test title helps understanding why and when this was added (faster than blame).

File: tests/testthat/test-colwise-select.R [hunk 2]
```diff
@@ -71,3 +71,9 @@ test_that("select_if() handles quoted predicates", {
...
+  df <- data.frame(a = 1, b = 2)
+  out <- df %>% group_by(a) %>% rename_all(toupper)
+  expect_identical(out, group_by(data.frame(a = 1, B = 2), a))
```
Comment (krlmlr) [inline tests/testthat/test-colwise-select.R]: Do we need to update documentation to describe the behavior?

Comment (lionel-) [reply tests/testthat/test-colwise-select.R]: I don't know what should be the behaviour, cf my comment in the PR description. cc @hadley

Comment (krlmlr) [reply tests/testthat/test-colwise-select.R]: In the first post in this PR, you're suggesting that the new behavior is consistent with `mutate_all()` and `summarize_all()`. You added a test, too, this qualifies as defined behavior. I think we may as well document it the way it is implemented and tested now.

**Comment by krlmlr** [comment]
Thanks!

File: tests/testthat/test-colwise-select.R (hunk 2 above)
Comment (lionel-) [reply tests/testthat/test-colwise-select.R]: @krlmlr I'm not sure why but I'm not systematically getting email notifications, I missed that comment. I still think that it makes sense for names operations to apply on the whole tibble and not just the grouped columns.

**Pull request merged**